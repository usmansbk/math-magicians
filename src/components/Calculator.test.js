import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import CalculatorPage, { Calculator } from './Calculator';

describe('Calculator', () => {
  it('renders page correctly', () => {
    const tree = renderer.create(<CalculatorPage />);

    expect(tree).toMatchSnapshot();
  });

  it('displays result of 8.2 รท 2 = 4.1', () => {
    const { getByText } = render(<Calculator />);

    fireEvent.click(getByText('8'));
    fireEvent.click(getByText('.'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('รท'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));

    expect(getByText('4.1')).toBeInTheDocument();
  });
});
