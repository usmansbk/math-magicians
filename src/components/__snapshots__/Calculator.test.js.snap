// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders page correctly 1`] = `
<section
  className="calculator-container"
>
  <h2
    className="section-title"
  >
    Let's do some math
  </h2>
  <div
    className="calculator-content"
  >
    <div
      className="calculator"
    >
      <div
        className="output"
      >
        <p>
          0
        </p>
      </div>
      <div
        className="grid"
      >
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="AC"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="+/-"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="%"
        />
        <input
          className="button grid-item bg-orange"
          onClick={[Function]}
          type="button"
          value="รท"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="7"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="8"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="9"
        />
        <input
          className="button grid-item bg-orange"
          onClick={[Function]}
          type="button"
          value="x"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="4"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="5"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="6"
        />
        <input
          className="button grid-item bg-orange"
          onClick={[Function]}
          type="button"
          value="-"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="1"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="2"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="3"
        />
        <input
          className="button grid-item bg-orange"
          onClick={[Function]}
          type="button"
          value="+"
        />
        <input
          className="button grid-item span-2-col"
          onClick={[Function]}
          type="button"
          value="0"
        />
        <input
          className="button grid-item "
          onClick={[Function]}
          type="button"
          value="."
        />
        <input
          className="button grid-item bg-orange"
          onClick={[Function]}
          type="button"
          value="="
        />
      </div>
    </div>
  </div>
</section>
`;
